mewarpx/mewarpx/emission.py

```diff
@@ -0,0 +1,877 @@
+        Notes:
+            Must be run after w3d dimensions (warp.w3d.xmmin etc) and cell
+            numbers (warp.w3d.nx etc) are set.
+        Arguments:
+            emitter (:class:`metools.emission.Emitter`): Emitter object that
+                will specify positions and velocities of particles to inject.
+            species (mepicmi.Species): A premade species. Note only electrons
+                will actually give physically meaningful weight calculations.
+            npart_per_cellstep (int): Number of macroparticles to inject per
+                cell on the cathode surface per timestep
+            T (float): Cathode temperature (K)
+            WF (float): Cathode work function (eV)
+            A (float): Coefficient of emission in Amp/m^2/K^2. Default 1.2e6.
+            allow_poisson (bool): If True and < npart_per_cellstep electrons
+                would be injected per cell; inject whole electrons with a
+                Poisson distribution. If False, inject fractions of electrons.
+                Default False.
+            wfac (float): Constant factor applied to variable particle
+                weights, which changes the actual injection weight from the
+                physically calculated quantity. Currently used only for
+                testing, or for e.g.  artificially lowering weight of trace
+                particles.
+            name (str or None): Injector name for diagnostics. Constructed from
+                speciesname if not given.
+            profile_decorator (decorator): A decorator used to profile the
+                injection methods and related functions.
+            unique_particles (bool): Whether WarpX will keep all particles
+                given it from every processor (True) or keep only a fraction of
+                particles based on processor count (False).
+        """
+        # Save class parameters
+        self.emitter = emitter
+        self.species = species
+        self.T = T
+        self.WF = WF
+        self.A = A
+        self.wfac = wfac
+
+        if profile_decorator is not None:
+            self.inject_particles = profile_decorator(self.inject_particles)
+            self.record_injectedparticles = (
+                profile_decorator(self.record_injectedparticles)
+            )
+
+        self.name = name
+        if self.name is None:
+            self.name = "thermionic_injector_" + self.species.name
+        self.unique_particles = unique_particles
+
+        area = self.emitter.area
+        dt = mwxrun.get_dt()
+        if (area is None) or (area <= 0.0) or (dt <= 0.0):
+            raise ValueError(
+                ("area {} or dt {} is invalid for injection.").format(
+                    area, dt))
+
+        # Determine weight and injection numbers
+        electrons_per_step = (minutil.J_RD(self.T, self.WF, self.A)
+                              * area * dt / picmi.constants.q_e)
```

Comment:
For the sake of keeping this public repo not dependent on private `minerva` we should probably just reimplement `J_RD` in mewarpx.util?